### ISSUE TITLE
Distributed API: No way to analyze URLs?

### ISSUE THREAD

**Issue by MarkusTeufelberger** [cuckoosandbox/cuckoo#1232]
According to http://docs.cuckoosandbox.org/en/latest/usage/dist/#post-api-task "POST /api/task - Submit a new file or URL to be analyzed".

https://github.com/cuckoosandbox/cuckoo/blob/master/distributed/distributed/views/api.py#L186 seems to error out as soon as no actual file is provided though...

**Comment by jbremer** [comment]
That's correct. Feel free to do a PR though!

**Comment by MarkusTeufelberger** [comment]
Every db.Task has a filename, should I overload this if an URL is checked making the name wrong or should there be an extra field (which would mean that the db schema would be updated)?

**Comment by MarkusTeufelberger** [comment]
I can do a PR, but I'll need to know first if I should extend your database schema or not. I personally would just add the URL column...

**Comment by jbremer** [comment]
I guess that'd be fine, yeah :) Sorry for the delay in answering.

**Comment by jbremer** [comment]
Did you have any progress on this feature @MarkusTeufelberger? Unfortunately I did too many database migrations recently, so I guess I could also take up on this issue if you'd really want me to.

**Comment by MarkusTeufelberger** [comment]
Yeah, sorry for not putting up a PR right away, I'd be more than happy if you are taking this one up.
